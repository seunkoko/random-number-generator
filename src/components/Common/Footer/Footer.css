.footer {
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  text-align: center;
  background-color: #DDDDDD;
  z-index: 10;
}

.footer__container {
  padding: 1rem;
  height: 3rem;
  font-size: 0.4rem;
}

.text-muted {
  color: #222222ba;
  letter-spacing: 0.2rem;
}