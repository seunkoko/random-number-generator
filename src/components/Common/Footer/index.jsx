import React from 'react';

// styles
import './Footer.css';

/**
 * Footer - Footer component
 *
 * @class Footer
 */
const Footer = () => (
  <footer className="footer">
    <div className="container">
      <span className="text-muted">Â© seunkoko 2018</span>
    </div>
  </footer>
);

export default Footer;
