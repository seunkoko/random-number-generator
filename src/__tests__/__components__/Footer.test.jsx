// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import Footer from '../../components/Common/Footer';

describe('Footer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  it('renders properly', () => {
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('span').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
