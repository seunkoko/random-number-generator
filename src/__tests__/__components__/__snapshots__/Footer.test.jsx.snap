// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="footer__container"
>
        <span
                className="text-muted"
        >
                © seunkoko 2018
        </span>
</div>,
      "className": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="text-muted"
>
          © seunkoko 2018
</span>,
        "className": "footer__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© seunkoko 2018",
          "className": "text-muted",
        },
        "ref": null,
        "rendered": "© seunkoko 2018",
        "type": "span",
      },
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="footer__container"
>
          <span
                    className="text-muted"
          >
                    © seunkoko 2018
          </span>
</div>,
        "className": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="text-muted"
>
            © seunkoko 2018
</span>,
          "className": "footer__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "© seunkoko 2018",
            "className": "text-muted",
          },
          "ref": null,
          "rendered": "© seunkoko 2018",
          "type": "span",
        },
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
